Sources/XcodeGenKit/PBXProjGenerator.swift

```diff
@@ -910,37 +910,45 @@ public class PBXProjGenerator {
         return "\(carthagePath)/\(platformName)"
     }
 
-    func getAllCarthageDependencies(target: Target) -> [Dependency] {
+    func getAllCarthageDependencies(target topLevelTarget: Target) -> [Dependency] {
         // this is used to resolve cyclical target dependencies
         var visitedTargets: Set<String> = []
         var frameworks: [String: Dependency] = [:]
 
-        var queue: [Target] = [target]
+        var queue: [ProjectTarget] = [topLevelTarget]
         while !queue.isEmpty {
-            let target = queue.removeFirst()
-            if visitedTargets.contains(target.name) {
+            let projectTarget = queue.removeFirst()
+            if visitedTargets.contains(projectTarget.name) {
                 continue
             }
-
-            for dependency in target.dependencies {
-                // don't overwrite frameworks, to allow top level ones to rule
-                if frameworks.contains(reference: dependency.reference) {
-                    continue
+            
+            if let target = projectTarget as? Target {
+                for dependency in target.dependencies {
+                    // don't overwrite frameworks, to allow top level ones to rule
+                    if frameworks.contains(reference: dependency.reference) {
+                        continue
+                    }
+                    
+                    switch dependency.type {
+                    case .carthage:
+                        frameworks[dependency.reference] = dependency
+                    case .target:
+                        if let projectTarget = project.getProjectTarget(dependency.reference) {
+                            queue.append(projectTarget)
+                        }
+                    default:
+                        break
+                    }
                 }
-
-                switch dependency.type {
-                case .carthage:
-                    frameworks[dependency.reference] = dependency
-                case .target:
-                    if let target = project.getTarget(dependency.reference) {
-                        queue.append(target)
+            } else if let aggregateTarget = projectTarget as? AggregateTarget {
+                for dependencyName in aggregateTarget.targets {
+                    if let projectTarget = project.getProjectTarget(dependencyName) {
```

Comment:
This highlights a question. Should `getAllDependenciesPlusTransitiveNeedingEmbedding` be checking `aggregateTarget.targets` as well?

Reply:
I went down that route at first. It wouldn't since it's the same as `shouldEmbedDependencies` in my mind. An aggregate target will be merging it's targets, they shouldn't propagate up, at least automatically. If that's the desire, it can still be done manually.

In the new algorithm this will be expressible explicitly.

Reply:
👍